alf/algorithms/algorithm.py

```diff
@@ -176,6 +179,11 @@ def set_path(self, path):
     def on_policy(self):
         return self._is_on_policy
 
+    @property
+    def has_offline(self):
+        # whether has offline data
+        return self._has_offline
```

Comment:
`self._has_offline` is undefined for non-RL algorithm

Reply:
Changed to return False for non-RL algorithms.